tests/test_spark_cni.py

```diff
@@ -44,65 +45,158 @@ def configure_security():
 
 
-@pytest.fixture(scope='module')
+@pytest.fixture()
 def setup_spark(configure_security, configure_universe):
     try:
         utils.upload_dcos_test_jar()
         utils.require_spark()
-        sdk_cmd.run_cli('package install --cli dcos-enterprise-cli --yes')
         yield
     finally:
         utils.teardown_spark()
 
 
+# The following dispatcher fixtures rely on sdk_install.install because for the time being
+# spark_utils.require_spark can't be effectively used with virtual network due to
+# https://jira.mesosphere.com/browse/DCOS-45468 (Add supporting endpoints for services
+# running in CNI (e.g. Calico) to AdminRouter)
+@pytest.fixture()
+def spark_dispatcher(configure_security, configure_universe, use_ucr_containerizer):
+    sdk_install.uninstall(utils.SPARK_PACKAGE_NAME, CNI_DISPATCHER_SERVICE_NAME, zk=CNI_DISPATCHER_ZK)
+
+    options = {
+        "service": {
+            "UCR_containerizer": use_ucr_containerizer
+        }
+    }
+
+    try:
+        merged_options = sdk_install.merge_dictionaries(CNI_SERVICE_OPTIONS, options)
+        sdk_install.install(
+            utils.SPARK_PACKAGE_NAME,
+            CNI_DISPATCHER_SERVICE_NAME,
+            0,
+            additional_options=utils.get_spark_options(CNI_DISPATCHER_SERVICE_NAME, merged_options),
+            wait_for_deployment=False)
+        yield
+    finally:
+        sdk_install.uninstall(utils.SPARK_PACKAGE_NAME, CNI_DISPATCHER_SERVICE_NAME, zk=CNI_DISPATCHER_ZK)
+
+
+@pytest.mark.sanity
+@pytest.mark.parametrize('use_ucr_containerizer', [False])
+def test_cni_dispatcher_docker(spark_dispatcher):
+    task = _get_dispatcher_task()
+    _check_task_network(task, is_docker=True)
+
+
 @pytest.mark.sanity
-def test_cni(setup_spark):
-    utils.run_tests(app_url=utils.SPARK_EXAMPLES,
-                    app_args="",
-                    expected_output="Pi is roughly 3",
-                    args=["--conf spark.mesos.network.name=dcos",
-                          "--class org.apache.spark.examples.SparkPi"])
+@pytest.mark.parametrize('use_ucr_containerizer', [True])
+def test_cni_dispatcher_ucr(spark_dispatcher):
+    task = _get_dispatcher_task()
+    _check_task_network(task)
 
 
 @pytest.mark.sanity
 @pytest.mark.smoke
 def test_cni_labels(setup_spark):
     driver_task_id = utils.submit_job(app_url=utils.SPARK_EXAMPLES,
                                       app_args="3000",   # Long enough to examine the Driver's & Executor's task infos
-                                      args=["--conf spark.mesos.network.name=dcos",
+                                      args=["--conf spark.mesos.network.name={}".format(NETWORK_NAME),
                                             "--conf spark.mesos.network.labels={}".format(SPARK_NETWORK_LABELS),
                                             "--conf spark.cores.max={}".format(CNI_TEST_NUM_EXECUTORS),
+                                            "--conf spark.mesos.containerizer=mesos",
```

Comment:
This test passed before w/o ucr, is this change intentional?

Reply:
yes because Docker doesn't support network labels